sklearn/base.py

```diff
@@ -416,20 +421,28 @@ def _validate_data(self, X, y='no_validation', reset=True,
         Returns
         -------
         out : {ndarray, sparse matrix} or tuple of these
-            The validated input. A tuple is returned if `y` is not None.
+            The validated input. A tuple is returned if both `X` and `y` are
+            validated.
         """
+        if y is None and self._get_tags()['requires_y']:
+            raise ValueError(
+                f"This {self.__class__.__name__} estimator "
+                f"requires y to be passed, but the target y is None."
+            )
 
-        if y is None:
-            if self._get_tags()['requires_y']:
-                raise ValueError(
-                    f"This {self.__class__.__name__} estimator "
-                    f"requires y to be passed, but the target y is None."
-                )
-            X = check_array(X, **check_params)
-            out = X
-        elif isinstance(y, str) and y == 'no_validation':
+        no_val_X = isinstance(X, str) and X == 'no_validation'
+        no_val_y = y is None or isinstance(y, str) and y == 'no_validation'
+
+        if no_val_X and no_val_y:
+            raise ValueError("Validation should be done on X, y or both.")
+        elif not no_val_X and no_val_y:
             X = check_array(X, **check_params)
             out = X
+        elif no_val_X and not no_val_y:
+            multi_output = check_params.get('multi_output', False)
+            y_numeric = check_params.get('y_numeric', False)
```

Comment:
I prefer passing in `check_params` directly into `_check_y` and include the following in the docstring:

> When validating only `y`, the only accepted `check_params` are multi_output and y_numeric.

This means an error will be raised when an unknown `kwarg` is passed in (such as `copy=True`). What do you think?

Reply:
Sounds good to me. @jeremiedbb I let you do the change if you agree :)